src/include/homestore/meta_service.hpp

```diff
@@ -121,7 +125,7 @@ class MetaBlkService {
      * @param cb : subsystem cb
      */
     void register_handler(meta_sub_type type, const meta_blk_found_cb_t& cb, const meta_blk_recover_comp_cb_t& comp_cb,
```

Comment:
It needs to return bool if the dep requested can cause circle or any invalidation to the graph.